scene/gui/licenses_dialog.cpp

```diff
@@ -0,0 +1,144 @@
+/**************************************************************************/
+/*  licenses_dialog.cpp                                                   */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "licenses_dialog.h"
+
+#include "core/license.gen.h"
+#include "rich_text_label.h"
+#include "scene/resources/style_box_flat.h"
+#include "scene/scene_string_names.h"
+
+void LicensesDialog::_close_button_pressed() {
+	SceneTree::get_singleton()->set_licenses_dialog_visible(false);
+}
+
+void LicensesDialog::_bind_methods() {
+}
+
+void LicensesDialog::unhandled_input(const Ref<InputEvent> &p_event) {
+	if (p_event->is_action_pressed(SNAME("ui_cancel"), false, true)) {
+		SceneTree::get_singleton()->set_licenses_dialog_visible(false);
+		Object::cast_to<Node>(this)->get_viewport()->set_input_as_handled();
```

Comment:
Is there a cleaner way of calling Node's `get_viewport()` (which returns a `Viewport *`) instead of CanvasLayer's `get_viewport()` which returns a RID?

Reply:
You could try `Node::get_viewport()`